### PR TITLE
Display Error as string in JSON diff tab

### LINKED ISSUE #42: Uncaught TypeError: Cannot read property 'length' of undefined
I am working with the [redux-actions](https://github.com/acdlite/redux-actions) library, which are utilities for creating [FSA-compliant actions](https://github.com/acdlite/flux-standard-action). If an `Error` is supplied to the action creator, an expected action would look like this:

``` js
const increment = createAction('INCREMENT');

const error = new TypeError('not a number');

expect(increment(error)).to.deep.equal({
  type: 'INCREMENT',
  payload: error,
  error: true
});
```

The `redux-devtools-inspector` is having trouble diffing the `Error` payload if it is added to the application's Redux state and throws:

```
Uncaught TypeError: Cannot read property 'length' of undefined
```

At [this line](https://github.com/alexkuz/redux-devtools-inspector/blob/9ce7985f16af9205013bac0b87e0fbe9a62aef55/src/tabs/JSONDiff.jsx#L12) in `src/tabs/JSONDiff.jsx`.

I looked into the [javascript-stringify](https://github.com/blakeembrey/javascript-stringify) function, and it returns `undefined` when supplied an `Error()` - as a result, calling `str.length` in the JSON diff component throws an error. (I am thinking this might be a bug in that library as well? I'm not sure, but can open an issue.)

The quickest way to reproduce this was to update the demo app in this repo to add a reducer that returns an `Error`.

``` diff
diff --git a/demo/src/js/DemoApp.jsx b/demo/src/js/DemoApp.jsx
index 4c4b572..889c75d 100644
--- a/demo/src/js/DemoApp.jsx
+++ b/demo/src/js/DemoApp.jsx
@@ -145,6 +145,9 @@ class DemoApp extends React.Component {
             <Button onClick={this.props.addSymbol} style={styles.button}>
               Add Symbol
             </Button>
+            <Button onClick={this.props.addError} style={styles.button}>
+              Add Error
+            </Button>
           </div>
         </div>
         <div style={styles.links}>
@@ -204,6 +207,7 @@ export default connect(
     }),
     addFunction: () => ({ type: 'ADD_FUNCTION' }),
     addSymbol: () => ({ type: 'ADD_SYMBOL' }),
+    addError: () => ({ type: 'ADD_ERROR' }),
     pushRoute
   }
 )(DemoApp);
diff --git a/demo/src/js/reducers.js b/demo/src/js/reducers.js
index 54ca2d5..974499f 100644
--- a/demo/src/js/reducers.js
+++ b/demo/src/js/reducers.js
@@ -91,5 +91,7 @@ export default {
   addFunction: (state=null, action) => action.type === 'ADD_FUNCTION' ?
     { f: FUNC } : state,
   addSymbol: (state=null, action) => action.type === 'ADD_SYMBOL' ?
-    { s: Symbol('symbol') } : state
+    { s: Symbol('symbol') } : state,
+  addError: (state=null, action) => action.type === 'ADD_ERROR' ?
+    Error() : state
 };
```

I will open a PR with a potential fix for this repo, but perhaps a patch is also needed in javascript-stringify? What do you think?

(Thanks for the awesome tool! ❤️)

### PR DESCRIPTION
**Pull request by macklinu**
Fixes #42 

redux-devtools-inspector-error

### CODE REVIEW

File: src/tabs/JSONDiff.jsx [hunk 1]
```diff
@@ -5,10 +5,8 @@ import getItemString from './getItemString';
 import getJsonTreeTheme from './getJsonTreeTheme';
 
 function stringifyAndShrink(val) {
+  if (val instanceof Error) { return `${val.toString()}()`; }
   const str = stringify(val);
-  if (val === null) { return 'null'; }
-  else if (typeof val === 'undefined') { return 'undefined'; }
```
Comment (macklinu) [inline src/tabs/JSONDiff.jsx:10]: I believe the stringify function should take care of these cases:

```
> const stringify = require('javascript-stringify')
undefined
> stringify(null)
'null'
> stringify(undefined)
'undefined'
```

**Comment by zalmoxisus** [comment]
I guess it will not work when except `Error()` you'll have other values changed.

**Comment by macklinu** [comment]
@zalmoxisus I'm not sure I follow your comment - can you please explain further?

**Comment by zalmoxisus** [comment]
@macklinu, `stringifyAndShrink` function is used for diffs. In your case the difference is an Error object, which passes the condition. However, if after dispatching an action, you change not only the `error` key, the  condition won't pass (`stringifyAndShrink`'s argument will not be an instance of `Error`).

**Comment by macklinu** [comment]
I see where you're coming from - thanks for the explanation @zalmoxisus. Let me take a step back and see if I can get a more specific use case I'm seeing in my application where this `stringifyAndShrink` function is throwing a `TypeError` and open a different PR / update #42 with more details.

**Comment by zalmoxisus** [comment]
Thanks anyway for the contribution! It would be better to transform Error objects on [`javascript-stringify`](https://github.com/blakeembrey/javascript-stringify) part.

**Pull request closed without merging**